src/execution.rs

```diff
@@ -320,16 +411,51 @@ impl Stanza {
                 shorthands: shorthands,
             };
             for statement in &self.statements {
-                statement
-                    .execute(&mut exec)
-                    .with_context(|| format!("Executing {}", statement))?;
+                statement.execute(&mut exec).or_else(|e| {
+                    Err(e).with_context(|| {
+                        let node = mat
+                            .captures
+                            .iter()
+                            .find(|c| c.index as usize == self.full_match_file_capture_index)
```

Comment:
This could be replaced by a call to this helper function: https://github.com/tree-sitter/tree-sitter-graph/blob/main/src/execution.rs#L955-L986.

Reply:
Thank you!